core/src/banking_stage/transaction_scheduler/transaction_state_container.rs

```diff
@@ -206,6 +196,129 @@ impl<Tx: TransactionWithMeta> TransactionStateContainer<Tx> {
     }
 }
 
+pub type SharedBytes = Arc<Vec<u8>>;
+
+/// A wrapper around `TransactionStateContainer` that allows re-uses
+/// pre-allocated `Bytes` to copy packet data into and use for serialization.
+/// This is used to avoid allocations in parsing transactions.
+pub struct TransactionViewStateContainer {
+    inner: TransactionStateContainer<RuntimeTransaction<ResolvedTransactionView<SharedBytes>>>,
+    bytes_buffer: Box<[SharedBytes]>,
+}
+
+pub(crate) struct SuccessfulInsert {
+    pub state: TransactionState<RuntimeTransaction<ResolvedTransactionView<SharedBytes>>>,
+}
+
+impl TransactionViewStateContainer {
+    /// Returns true if packet was dropped due to capacity limits.
+    pub(crate) fn try_insert_with_data(
+        &mut self,
+        data: &[u8],
+        f: impl FnOnce(SharedBytes) -> Result<SuccessfulInsert, ()>,
+    ) -> bool {
+        if self.inner.id_to_transaction_state.len() == self.inner.id_to_transaction_state.capacity()
+        {
+            return true;
+        }
+        // Get remaining capacity before inserting.
+        let remaining_capacity = self.remaining_capacity();
+
+        // Get a vacant entry in the slab.
+        let vacant_entry = self.inner.id_to_transaction_state.vacant_entry();
+        let transaction_id = vacant_entry.key();
+
+        // Get the vacant space in the bytes buffer.
+        let bytes_entry = &mut self.bytes_buffer[transaction_id];
+        // Assert the entry is unique, then copy the packet data.
+        {
+            assert_eq!(Arc::strong_count(bytes_entry), 1, "entry must be unique");
+            let bytes = Arc::make_mut(bytes_entry);
+
+            // Clear and copy the packet data into the bytes buffer.
+            bytes.clear();
+            bytes.extend_from_slice(data);
+        }
+
+        // Attempt to insert the transaction, storing the frozen bytes back into bytes buffer.
```

Comment:
Forgot to remove the comment when I updated the code to not use `Bytes`. Originally, this had a state transition in an enum where we need to move the frozen bytes back into `bytes_buffer`.

It's 100% simpler this way, and I can just delete this comment since it is just entirely wrong.